include/seastar/core/chunked_fifo.hh

```diff
@@ -184,6 +189,10 @@ public:
     inline void pop_front() noexcept;
     inline bool empty() const noexcept;
     inline size_t size() const noexcept;
+    // Pop the first n elements from the fifo. Equivalent to calling pop_front()
+    // n times, though likely to be faster for n greater than 1. The fifo must
+    // contain at least n elements or the behavior is undefined.
+    void pop_front_n(size_t n) noexcept;
```

Comment:
Would `void pop_front(size_t n = 1) noexcept` look nicer?

Reply:
I think it would pessimize `pop_front()` which has to pay for the extra complexity it doesn't need but it could be tested, but we should wait for the resolution of this [other discussion](https://github.com/scylladb/seastar/pull/2357/files#r1688243424) since it impacts it.